### PR TITLE
Adding tests for READY and ALIVE

### PR DESCRIPTION
**Pull request by matzew**
# Changes

- :broom: Add tests

/kind <kind>

Fixes #

**Release Note**

```release-note

```

**Docs**

```docs

```

### CODE REVIEW

File: cloudevents/service_test.go [hunk 1]
```diff
@@ -290,3 +292,107 @@ func TestHandle_Invoked(t *testing.T) {
...
+		t.Fatalf("unexpected http status code: %v", resp.StatusCode)
+	}
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if string(body) != "ALIVE" {
+		t.Fatalf("unexpected body: %v\n", string(body))
+	}
+
```
Comment (lkingland) [inline cloudevents/service_test.go:397]: I don't think it's necessary to check the reponse body.  That is very much something that could change easily.  The important one is the 200OK

**Review by lkingland — COMMENTED** [review]
I like that we have the tests which ensure the default handler is created and exposed for both the readiness and liveness checks.

Do we want to add tests now which ensure that a function which overrides these checks are invoked and respected?

I'd just remove the check for the exact string content of the response.  I could see us wanting to change that for various reasons (perhaps returning diagnostics if a specific debug header is set).  The important thing for _these_ checks is that it respnds 200OK

It might be worth adding a channel and a timeout as well, so that if something does go wrong, it fails the test quickly rather than wait for the test timeout.

**Comment by matzew** [comment]
```
Do we want to add tests now which ensure that a function which overrides these checks are invoked and respected? Maybe that's best left for a future PR.
```

later / separate sounds good

**Comment by matzew** [comment]
> add a timeout.

I can do that separately for all tests.

**Review by lkingland — APPROVED** [review]

**Pull request merged**